src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -1487,24 +1495,17 @@ static void getDeclaredUserDefinedBinaryOperatorsInScope(ArrayBuilder<NamedTypeS
                     }
                     else
                     {
-                        getDeclaredUserDefinedBinaryOperators(extensionDeclarationsInSingleScope, kind, name2Opt, operators);
+                        getDeclaredUserDefinedBinaryOperators(extensionCandidatesInSingleScope, kind, name2Opt, operators);
                     }
                 }
             }
 
-            static void getDeclaredUserDefinedBinaryOperators(ArrayBuilder<NamedTypeSymbol> extensionDeclarationsInSingleScope, BinaryOperatorKind kind, string name, ArrayBuilder<BinaryOperatorSignature> operators)
+            static void getDeclaredUserDefinedBinaryOperators(ArrayBuilder<Symbol> extensionCandidatesInSingleScope, BinaryOperatorKind kind, string name, ArrayBuilder<BinaryOperatorSignature> operators)
             {
-                foreach (NamedTypeSymbol extensionDeclaration in extensionDeclarationsInSingleScope)
-                {
-                    Debug.Assert(extensionDeclaration.IsExtension);
-
-                    if (extensionDeclaration.ExtensionParameter is null)
```

Comment:
>if (extensionDeclaration.ExtensionParameter is null) [](http://example.com/codeflow?start=20&length=52)

Where this check is performed now?

Reply:
`NamedTypeSymbol.GetExtensionMembers(...)`